src/main/java/com/netgrif/application/engine/elastic/service/ElasticTaskQueueManager.java

```diff
@@ -0,0 +1,166 @@
+package com.netgrif.application.engine.elastic.service;
+
+import com.netgrif.application.engine.elastic.domain.ElasticJob;
+import com.netgrif.application.engine.elastic.domain.ElasticTask;
+import com.netgrif.application.engine.elastic.domain.ElasticTaskJob;
+import com.netgrif.application.engine.elastic.domain.ElasticTaskRepository;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.dao.InvalidDataAccessApiUsageException;
+import org.springframework.scheduling.concurrent.ThreadPoolTaskExecutor;
+import org.springframework.stereotype.Service;
+
+import javax.annotation.PreDestroy;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.*;
+
+@Slf4j
+@Service
+public class ElasticTaskQueueManager {
+
+    private final ThreadPoolTaskExecutor elasticTaskExecutor;
+
+    private final ElasticTaskRepository repository;
+
+    private final ConcurrentHashMap<String, BlockingQueue<Runnable>> taskQueues = new ConcurrentHashMap<>();
+    private final Set<String> activeTasks = ConcurrentHashMap.newKeySet();
+
+    public ElasticTaskQueueManager(@Qualifier("elasticTaskExecutor") ThreadPoolTaskExecutor elasticTaskExecutor, ElasticTaskRepository repository) {
+        this.elasticTaskExecutor = elasticTaskExecutor;
+        this.repository = repository;
+    }
+
+
+    public Future<ElasticTask> scheduleOperation(ElasticTaskJob task) {
+        if (task.getTask().getTaskId() == null) {
+            throw new IllegalArgumentException("Task id cannot be null");
+        }
+
+        String taskId = task.getTask().getTaskId();
+        log.debug("Scheduling operation for task: {}", taskId);
+
+        CompletableFuture<ElasticTask> future = new CompletableFuture<>();
+        Runnable taskWrapper = () -> {
+            try {
+                ElasticTask result = processTask(task);
+                future.complete(result);
+            } catch (Exception e) {
+                future.completeExceptionally(e);
+            }
+        };
+
+        BlockingQueue<Runnable> queue = taskQueues.computeIfAbsent(taskId, k -> new LinkedBlockingQueue<>());
+        queue.add(taskWrapper);
```

Comment:
according to https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/BlockingQueue.html calling `.add()` might throw an exception, which is not caught. Is this intended?

Reply:
Done